Client/Frontend/Browser/Playlist/Controllers/PlaylistMoveFolderView.swift

```diff
@@ -0,0 +1,269 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Combine
+import Data
+import CoreData
+import Shared
+import BraveShared
+
+private struct PlaylistFolderImage: View {
+    static let cornerRadius = 5.0
+    private static let favIconSize = 16.0
+    private var thumbnailLoader: ImageLoader
+    private var favIconLoader: ImageLoader
+    
+    @State private var thumbnail = UIImage()
+    @State private var favIcon = UIImage()
+    
+    init(item: PlaylistItem) {
+        thumbnailLoader = ImageLoader(thumbnail: item)
+        favIconLoader = ImageLoader(favIcon: item)
+    }
+    
+    var body: some View {
+        ZStack(alignment: .topLeading) {
+            Image(uiImage: thumbnail)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 100.0, height: 60.0)
+                .background(Color.black)
+                .clipShape(RoundedRectangle(cornerRadius: PlaylistFolderImage.cornerRadius, style: .continuous))
+            
+            Image(uiImage: favIcon)
+                .resizable()
+                .aspectRatio(1.0, contentMode: .fit)
+                .frame(width: PlaylistFolderImage.favIconSize,
+                       height: PlaylistFolderImage.favIconSize)
+                .clipShape(RoundedRectangle(cornerRadius: 2.5, style: .continuous))
+                .padding(8.0)
+        }
+        .frame(width: 100.0, height: 60.0)
+        .onReceive(thumbnailLoader.$image) {
+            self.thumbnail = $0 ?? UIImage()
+        }
+        .onReceive(favIconLoader.$image) {
+            self.favIcon = $0 ?? UIImage()
+        }
+    }
+    
+    private class ImageLoader: ObservableObject {
+        @Published var image: UIImage?
+        
+        private let renderer = PlaylistThumbnailRenderer()
+        
+        init(thumbnail: PlaylistItem) {
+            guard let mediaSrc = thumbnail.mediaSrc,
+                  let assetUrl = URL(string: mediaSrc) else {
+                image = nil
+                return
+            }
+            
+            loadImage(url: assetUrl, isFavIcon: false)
+        }
+        
+        init(favIcon: PlaylistItem) {
+            guard let pageSrc = favIcon.pageSrc,
+                  let favIconUrl = URL(string: pageSrc) else {
+                image = nil
+                return
+            }
+            
+            loadImage(url: favIconUrl, isFavIcon: true)
+        }
+        
+        private func loadImage(url: URL, isFavIcon: Bool) {
+            renderer.loadThumbnail(assetUrl: isFavIcon ? nil : url,
+                                   favIconUrl: isFavIcon ? url : nil,
+                                   completion: { [weak self] image in
+                self?.image = image
+            })
+        }
+    }
+}
+
+private struct PlaylistFolderView: View {
+    let iconColor: Color
+    let titleColor: Color
```

Comment:
These are not needed. 

Pass in a boolean like `isSourceFolder` and based on that apply `.opacity(isSourceFolder ? 0.5 : 1.0)` to text and `.foregroundColor(isSourceFolder ? Color(.braveDisabled) : Color(.braveOrange))` to the icon